### PR TITLE
A less tightly-coupled approach to using the DB library and containers

### PR DESCRIPTION
**Pull request by borkweb**
This PR builds on top of the original PR (https://github.com/stellarwp/schema/pull/5) to add the DB library into the mix. However, the new approach builds on the configuration-based approach that @JasonTheAdams  and I talked about with the [stellarwp/db](https://github.com/stellarwp/db) library, allowing a project to inject the relevant values/objects. So, similarly, this library has two dependencies:

1. A container of some sort that conforms to a yet-to-be determined contract
1. The [stellarwp/db](https://github.com/stellarwp/db). But it isn't a composer dependency and most definitely not a Strauss included dependency

Those will get injected via `StellarWP\Schema\Config::set_[container|db]()` calls before `Schema::init()`.

I spent a while stripping the di52-specific API usage down to an approach that leverages the methods that it and the Give container share. Specifically:

* I removed the service provider fanciness that di52 provides and made it more bare-bones. The providers are set as singletons and the auto-called `register()` methods have been switched to `make()->register()` calls.
* I removed the `setVar()`/`getVar()` calls and replaced them with `bind( $string, $closure )` / `make()` calls instead

Note: I imagine further iterations on this code will be needed before we merge it fully.

### CODE REVIEW

File: src/Schema/Config.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace StellarWP\Schema;
+
+class Config {
+	/**
+	 * Container object.
+	 *
+	 * @var object
```
Comment (lucatume) [inline src/Schema/Config.php:9]: Going with the PSR-11 standard would mean this dependency type-hint would be `Psr\Container\ContainerInterface` from the [psr/container](https://packagist.org/packages/psr/container) Composer package.

**Review by lucatume — CHANGES_REQUESTED** [review]
Replace the dependency on the DI52 container with one on the PSR-11 specification.

File: README.md [hunk 2]
```diff
@@ -16,6 +16,15 @@ composer require stellarwp/schema
...
 > Luckily, adding Strauss to your `composer.json` is only slightly more complicated than adding a typical dependency, so checkout our [strauss docs](https://github.com/stellarwp/global-docs/blob/main/docs/strauss-setup.md).
 
+## Usage prerequisites
+
+To actually _use_ the schema library, you must have two additional libraries in your project:
+
+1. A Dependency Injection Container (DI Container) that is compatible with [di52](https://github.com/lucatume/di52) _(We recommend using di52.)_.
+1. The [stellarwp/db](https://github.com/stellarwp/db) library.
+
+In order to keep this library as light as possible, those dependcies are not included in the library itself. To avoid version compatibility issues, those libraries are _also_ not included as Composer dependencies. Instead, you must include them in your project. We recommend including them via composer [using Strauss](https://github.com/stellarwp/global-docs/blob/main/docs/strauss-setup.md), just like you have done with this library.
```
[the remark this answers is not in the document]
Comment (borkweb) [reply README.md:26]: If set as a dependency via composer, we'd have a version of stellarwp/db that'd conflict with other projects that have it as a dependency. If we use Strauss on it, we'd have multiple isolated versions. 

A project that attempts to use both schema AND db will need to go through extra hoops to override things. Like so:

```
use Bork\StellarWP\Schema\StellarWP\DB\Config as Schema_DB_Config;
use Bork\StellarWP\DB\Config as DB_Config;

Schema_DB_Config::setHookPrefix( 'bork' );
DB_Config::setHookPrefix( 'bork');
```

At one point I was thinking that the hook prefix and the DatabaseQueryException classes could additionally be set via filters, then we could get in a situation where multiple plugins that use libraries that use DB and/or DB directly all hook to the same filter and override each other's prefixes and exception classes leading to unpredictability. 

This was one case where explicitness seemed appropriate. I'm not tied to this line of thinking if there is an approach that suits the need while remaining flexible.

File: README.md [hunk 3]
```diff
@@ -16,6 +16,15 @@ composer require stellarwp/schema
 >
 > Luckily, adding Strauss to your `composer.json` is only slightly more complicated than adding a typical dependency, so checkout our [strauss docs](https://github.com/stellarwp/global-docs/blob/main/docs/strauss-setup.md).
 
+## Usage prerequisites
+
+To actually _use_ the schema library, you must have two additional libraries in your project:
+
+1. A Dependency Injection Container (DI Container) that is compatible with [di52](https://github.com/lucatume/di52) _(We recommend using di52.)_.
```
[the remark this answers is not in the document]
Comment (borkweb) [reply README.md:23]: PSR-11 doesn't require _enough_ - `get()` and `has()` is a pretty weak Interface, so what we should probably do is have a StellarWP container interface - one that requires the following:

- `bind()`
- `callback()`
- `get()`
- `has()`
- `singleton()`

**Pull request merged**